client/reader/stream/reader-tag-sidebar/index.jsx

```diff
@@ -1,14 +1,14 @@
 import { Button } from '@automattic/components';
+import formatNumber from '@automattic/components/src/number-formatters/lib/format-number';
```

Comment:
I'd generally recommend importing only public API. That means going with one of the following alternatives:

- Exporting `formatNumber` publicly from the package and consuming it from there
- Moving `formatNumber` to another package where it makes the most sense.

Since number formatting has little to do with components, I'd recommend the latter.

Reply:
Agreed. I've moved it to i18n-utils, where I'd argue it makes the most sense, since this is mostly about ensuring correct number formatting in all languages.

Reply:
Thanks. 

I'd like feedback @Automattic/i18n on this as well. Particularly because we already have a `numberFormat` utility in the `i18n-calypso` package and I'm not sure how much we want to duplicate.

Reply:
The part that could be concerning is the locale. If I understand correctly, we won't consider the user locale from the user profile, but only the browser one, which is often not the right one. I have serious doubts that this is what we want to do; I believe instead we want to use the locale from the user's chosen interface language.

Reply:
My understanding is that we pass the locale slug specifically to avoid this? For instance, this is me with my WP.com profile in Turkish, and my browser in English.

<img width="324" alt="Screenshot 2024-03-18 at 08 49 20" src="https://github.com/Automattic/wp-calypso/assets/43215253/4e5354aa-afb0-4752-8789-50b6f529b1cb">

Do note that in some cases, such as the above one, we're already using the browser for the number on the right but not the one on the left!

Reply:
Right. But in this PR I see a bunch of usages where we don't pass the slug specifically.

Reply:
Although I do wonder if it might be easier not to require the slug to be passed? `translate` does it, so it must somehow be possible (though haven't looked into it yet!).

Reply:
`translate()` is a different deal. It will use the internal i18n instance that gets updated once the slug changes, while the current formatting function you're working with won't do that - it will just statically use the default slug.

Reply:
Forgive me as I'm not quite sure I understand how this i18n works, but would it not be possible to just use the default slug like this?

https://github.com/Automattic/wp-calypso/blob/b9ea6f078e485fade468efc9daa10d43cdb4fc45/packages/i18n-utils/src/localize-url.tsx#L21-L23

Would that have unexpected problems?

Reply:
This leads to the same potential issue with using `getLocaleSlug()` directly that I pointed out in https://github.com/Automattic/wp-calypso/pull/87864#discussion_r1526489762.